### PR TITLE
feat(pack/java): add Java Language Pack

### LINKED ISSUE #75: Java Language Pack
I think that adding a java language pack (using nvim-jdtls) would be a great addition to the community packages.

The default recommended settings for jdtls should be a good example.

I don't currently have a sample, however the other language packs area already a good example, and nvim-jdtls has a solid base config.

It would also be good to install a valid formatter, as jdtls does not have very good formatting on it's own.

### PR DESCRIPTION
**Pull request by sjcobb2022**
Closes #75

Hi There,

Spent a part of the week making the java pack. The commit tree is really messy as I was using a git reference to the package to test it (probs not the best idea).

The package adds quite a few things so I'll list them here

- Adds `java` treesitter parsers
- Adds `html` treesitter parsers (XML)
- Adds `jdtls` language server
- Adds `lemminx` language server (XML)
- Adds [nvim-jdtls](https://github.com/mfussenegger/nvim-jdtls) for language specific tooling
- Adds `clang_format` through null-ls
- Adds `javadbg` and `javatest` debug adapters for nvim-dap

### CODE REVIEW

File: lua/astrocommunity/pack/java/java.lua [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                  .. "/extension/server/*.jar"
+                ),
+                "\n", {}))
+          },
+        },
+        handlers = {
+          ["language/status"] = function(_, result)
+            -- print(result)
+          end,
+          ["$/progress"] = function(_, result, ctx)
```
Comment (mehalter) [inline lua/astrocommunity/pack/java/java.lua:126]:
```suggestion
          ["$/progress"] = function()
```

Remove unused parameters

File: lua/astrocommunity/pack/java/java.lua [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+              vim.split(
+                vim.fn.glob(
+                  require("mason-registry").get_package("java-test"):get_install_path()
+                  .. "/extension/server/*.jar"
+                ),
+                "\n", {}))
+          },
+        },
+        handlers = {
+          ["language/status"] = function(_, result)
```
Comment (mehalter) [inline lua/astrocommunity/pack/java/java.lua:123]:
```suggestion
          ["language/status"] = function()
```

Remove unused parameters

**Review by mehalter — CHANGES_REQUESTED** [review]
Thank you so much for this! I made a couple comments to help the linter not be angry with you. Also please make sure to format the files with `stylua` and not the Lua language server, that way the style checks can pass as well

**Comment by mehalter** [comment]
We can resolve the git commits history with squash/merge as well so no worries about that

**Comment by sjcobb2022** [comment]
@mehalter Fixed!

**Review by mehalter — CHANGES_REQUESTED** [review]
Thanks! Very close now, just the stylua on the file to pass the formatting checker. Just make sure you aren't using the Lua language servers for formatting and only using stylua

**Comment by sjcobb2022** [comment]
> Thanks! Very close now, just the stylua on the file to pass the formatting checker. Just make sure you aren't using the Lua language servers for formatting and only using stylua

Yeh sorry I just used github to edit.

**Comment by mehalter** [comment]
Thanks so much @sjcobb2022 !

**Pull request merged**